src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -400,6 +402,27 @@ private static void EndRequest(SafeCurlMultiHandle multiHandle, IntPtr statePtr,
         {
             GCHandle stateHandle = GCHandle.FromIntPtr(statePtr);
             RequestCompletionSource state = (RequestCompletionSource)stateHandle.Target;
+
+            try
+            {
+                if (CURLcode.CURLE_OK == result && state.ResponseMessage.StatusCode != HttpStatusCode.Unauthorized && state.Handler.PreAuthenticate)
+                {
+                    ulong availedAuth;
+                    if (Interop.libcurl.curl_easy_getinfo(state.RequestHandle, CURLINFO.CURLINFO_HTTPAUTH_AVAIL, out availedAuth) == CURLcode.CURLE_OK)
+                    {
+                        state.Handler.AddCredentialToCache(state.RequestMessage.RequestUri, availedAuth, state.NetworkCredential);
+                    }
+                    // ignore errors
+                }
+            }
+            catch
+            {
+                // ignoring the exception in this case.
```

Comment:
What exception is this ignoring / where would it be coming from, AddCredentialToCache?  Can we be more specific in what kinds of exceptions we're catching rather than catching everything?